File: homestay-monolith/src/main/java/vn/homestay/room/Room.java
package vn.homestay.room;

import jakarta.persistence.*;
import java.math.BigDecimal;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "rooms")
@Getter @Setter @NoArgsConstructor
public class Room {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "name", nullable = false, length = 100)
    private String name;
    @Column(name = "max_guests", nullable = false)
    private Integer maxGuests;
    @Column(name = "quantity", nullable = false)
    private Integer quantity;
    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal pricePerNight;
    @Column(length = 2000)
    private String description;
    private String imageUrl;
    @ManyToOne(optional = false, fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    private RoomCategory category;
}

File: homestay-frontend/src/types/room.ts
export interface Room {
    id: number;
    name: string;
    maxGuests: number;
    quantity: number;
    availableToday: number;
    availableForStay?: number | null;
    pricePerNight: number;
    description?: string | null;
    imageUrl?: string | null;
    categoryId: number;
    categoryName: string;
}

export interface PagedResponse<T> {
    content: T[];
    totalElements: number;
    totalPages: number;
    number: number; // trang hiện tại (bắt đầu từ 0)
    size: number;
}
